### ISSUE TITLE
Unit conversion with `Decimal` sometimes introduces trailing zeros

### ISSUE THREAD

**Issue by tovrstra** [hgrecco/pint#1621]
The following example shows somewhat unexpected behavior:

```python
import pint
from decimal import Decimal
ureg = pint.UnitRegistry(non_int_type=Decimal)
a = ureg.Quantity("5.0 mV/m")
b = a.to_base_units()
print(a)
print(b)
print(a.magnitude.as_tuple())
print(b.magnitude.as_tuple())
```

This prints out

```
5.0 millivolt / meter
0.005000 kilogram * meter / ampere / second ** 3.000
DecimalTuple(sign=0, digits=(5, 0), exponent=-1)
DecimalTuple(sign=0, digits=(5, 0, 0, 0), exponent=-6)
```

The conversion to base units, in some cases, introduces extra trailing zeros.
This can be confusing as it gives the impression there are suddenly more significant digits.
When using `pint` in an educational setting, this is not ideal.
(The issue cannot be reproduced with all units, e.g. no problem when repeating this example with `"5.0 mV"`.)

If was able to fix this by changing `1.0` to `1` on the following line:

https://github.com/hgrecco/pint/blob/c07fc9b218ea39965d8d4492e52168fe8f8a644b/pint/util.py#L568

I'm not sure if that could break other things?

(Note that this is just a minor snag. So far `pint` has been super useful! Thanks!!)

**Comment by hgrecco** [comment]
Thanks for pointing this out (and your comments about Pint). I think your proposed solution should work. Why don't you fork it, try your change and let's see what the test suite says (which is quite thorough)?

**Comment by tovrstra** [comment]
@hgrecco I meant to bring up the following. I'll need some feedback to know how to proceed.

In the PR, I naively added some code to `conftest.py`:

https://github.com/hgrecco/pint/blob/b0faa1437ec9e9a203b06e589b64ecbb39364058/pint/testsuite/conftest.py#L51-L61

This makes it possible to parametrize a registry for a unit test, e.g.

https://github.com/hgrecco/pint/blob/b0faa1437ec9e9a203b06e589b64ecbb39364058/pint/testsuite/test_issues.py#L1020-L1024

I'm not sure if this fits in the way you're organizing unit tests? I noticed that the unit tests on other `non_int_types` use a class hierarchy instead, but it comes with comment:

https://github.com/hgrecco/pint/blob/b0faa1437ec9e9a203b06e589b64ecbb39364058/pint/testsuite/test_non_int.py#L15-L16

Is there a roadmap for writing tests for non_int_types in a specific way? I can sense that the class-based approach is not great, because it quickly becomes hard to follow, with relevant bits in different places.

(P.S. Congrats with the new release!)

**Comment by hgrecco** [comment]
The class based organization predates the current structure, as before we were using `unittest`. Later we moved to pytest. There is still work to do there, particularly reusing the registry to make testing faster.

My problem with your proposal is that this:
```python
 @pytest.mark.func_registry_args(non_int_type=decimal.Decimal) 
 def test_issue1621(func_registry): 
     print(func_registry.non_int_type) 
     digits = func_registry.Quantity("5.0 mV/m").to_base_units().magnitude.as_tuple()[1] 
     assert digits == (5, 0) 
```
is not really shorter nor more comfortable than:
```python
 def test_issue1621(): 
     ureg = pint.UnitRegistry(non_int_type=decimal.Decimal)
     digits = ureg.Quantity("5.0 mV/m").to_base_units().magnitude.as_tuple()[1] 
     assert digits == (5, 0) 
```

Actually, people have suggested to remove `func_registry`.

In any case, I am happy with any effort to improve testing.

**Comment by tovrstra** [comment]
Thanks for explaining. That all makes a lot of sense. I've changed the PR along the lines of what you suggested.
